frontend/tests/badges/cppCourseGraduateTest.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+/**
+ * Simple test for legacy user Badge
+ *
+ * @author RuizYugen
+ */
+class CppCourseGraduate extends \OmegaUp\Test\BadgesTestCase {
+    public function testCourseCpp() {
+        $courseAlias = 'introduccion_a_cpp';
+
+        //create course
+        $courseData = \OmegaUp\Test\Factories\Course::createCourseWithAssignments(
+            /*$nAssignments=*/ 2,
+            $courseAlias
+        );
+        $studentsInCourse = 2;
+
+        // Prepare assignment. Create four problems: The first three accept
+        // submissions and the last one does not.
+        $adminLogin = self::login($courseData['admin']);
+        $problems = [];
+        for ($i = 0; $i < 3; $i++) {
+            $problems[] = \OmegaUp\Test\Factories\Problem::createProblem();
+        }
+        $problems[] = \OmegaUp\Test\Factories\Problem::createProblem(
+            new \OmegaUp\Test\Factories\ProblemParams([
+                'languages' => '',
+            ])
+        );
+
+        // Problems 1 and 2 will be assigned to the first assignment, both have
+        // submissions. Problem 3 and 4 will be assigned to second assignment
+        foreach ($problems as $index => $problemData) {
+            $assignmentAliasIndex = ($index === 0 || $index === 1) ? 0 : 1;
+            \OmegaUp\Controllers\Course::apiAddProblem(new \OmegaUp\Request([
+                'auth_token' => $adminLogin->auth_token,
+                'course_alias' => $courseData['course_alias'],
+                'assignment_alias' => $courseData['assignment_aliases'][
+                    $assignmentAliasIndex
+                ],
+                'problem_alias' => $problemData['request']['problem_alias'],
+            ]));
+        }
```

Comment:
esto es un ejemplo del [anti-patrón for-if](https://devblogs.microsoft.com/oldnewthing/20111227-00/?p=8793). lo mejor que se puede hacer es separarlo en dos ciclos:

```suggestion
        foreach (array_slice($problems, 0, 2) as $problemData) {
            \OmegaUp\Controllers\Course::apiAddProblem(new \OmegaUp\Request([
                'auth_token' => $adminLogin->auth_token,
                'course_alias' => $courseData['course_alias'],
                'assignment_alias' => $courseData['assignment_aliases'][0],
                'problem_alias' => $problemData['request']['problem_alias'],
            ]));
        }
        foreach (array_slice($problems, 2, 2) as $problemData) {
            \OmegaUp\Controllers\Course::apiAddProblem(new \OmegaUp\Request([
                'auth_token' => $adminLogin->auth_token,
                'course_alias' => $courseData['course_alias'],
                'assignment_alias' => $courseData['assignment_aliases'][1],
                'problem_alias' => $problemData['request']['problem_alias'],
            ]));
        }
```